### PR TITLE
Log matched but unsettled order UIDs in current run loop

### PR DESCRIPTION
**Pull request by fleupold**
We have gotten increased user complaints that the time from order submission to settlement is taking too long. It's likely that we could speed up individual settlements if instead of merging them (risk of single one causing failure for other orders) we submitted them in parallel.

Before we implement this I'd like to get an idea how many orders are currently "delayed" a run loop.

### Test Plan
CI

### CODE REVIEW

File: solver/src/driver.rs [hunk 1]
```diff
@@ -258,6 +258,27 @@ impl Driver {
         }
     }
 
+    // Log the amount of orders that were matched but not settled in this runloop (effectively queued for the next one)
+    // Should help us to identify how much we can save by parallelizing execution.
+    fn report_matched_but_unsettled_orders<I: Iterator<Item = Settlement>>(
+        &self,
+        submitted: &Settlement,
+        all: I,
+    ) {
```
Comment (vkgnosis) [inline solver/src/driver.rs:267]:
```suggestion
    fn report_matched_but_unsettled_orders(
        &self,
        submitted: &Settlement,
        all: impl Iterator<Item = Settlement>,
    ) {
```

**Review by vkgnosis — APPROVED** [review]

**Review by vkgnosis — APPROVED** [review]

File: solver/src/driver.rs [hunk 2]
```diff
@@ -258,6 +258,27 @@ impl Driver {
...
+            .iter()
+            .map(|trade| trade.order.order_meta_data.uid)
+            .collect();
+        let all_matched: HashSet<_> = all
+            .flat_map(|solution| solution.trades().to_vec())
+            .map(|trade| trade.order.order_meta_data.uid)
+            .collect();
+        let matched_but_not_settled: HashSet<_> = all_matched.difference(&submitted).collect();
+        self.metrics
+            .orders_matched_but_not_settled(matched_but_not_settled.len())
```
Comment (bh2smith) [inline solver/src/driver.rs:279]: The doc string on this method says that we log these values, but the implementation only seems to include it in the metrics.... Do the metrics have a log statement in them?

**Review by bh2smith — APPROVED** [review]
All good, just wondering where the actual logging is taking place?

File: solver/src/driver.rs (hunk 1 above)
Comment (vkgnosis) [reply solver/src/driver.rs:267]: https://doc.rust-lang.org/stable/book/ch10-02-traits.html#traits-as-parameters

It means the same thing but is easier to write.

File: solver/src/driver.rs (hunk 2 above)
Comment (fleupold) [reply solver/src/driver.rs:279]: By logging I mean report to a prometheus metrics, which grafana collects from our pods periodically. Will update the documentation to clarify this.

**Review by nlordell — APPROVED** [review]

**Pull request merged**